lib/ansible/modules/network/iosxr/iosxr_banner.py

```diff
@@ -155,20 +237,20 @@ def main():
     warnings = list()
     check_args(module, warnings)
 
-    result = {'changed': False}
-    result['warnings'] = warnings
-
-    want = map_params_to_obj(module)
-    have = map_config_to_obj(module)
+    device_capabilities = get_device_capabilities(module)
+    network_api = device_capabilities.get('network_api')
 
-    commands = map_obj_to_commands((want, have), module)
-    result['commands'] = commands
-
-    if commands:
-        if not module.check_mode:
-            load_config(module, commands, result['warnings'], commit=True)
-        result['changed'] = True
+    if network_api == 'cliconf':
+        module.deprecate('cli support for \'iosxr_banner\' is deprecated. use netconf instead', version='4 releases from v2.5')
```

Comment:
Use double quotes, then you will not need to escape
Also think about wording, perhaps:
`"cli support for 'iosxr_banner' is deprecated. Use transport 'netconf'  instead", version='4 releases from v2.5`

Is version an int or string?

Reply:
Version is a string. Will fix the wordings.